### PR TITLE
Adds config option to disable wraparound

### PR DESCRIPTION
**Pull request by colinhanrahan**
While working on this, I overhauled agents' Von Neumann and Moore neighbors.
- Removed `agent.vonNeumannNeighbors` and `agent.mooreNeighbors`
- Removed `update*Neighbors()` functions
- Added `agent.socialNeighbors` because we still need asymmetrical adjacency neighborhoods for network overlay
- All other functions that used adjacent neighbors now use `agent.cell.neighbors.values()`
- `cell.neighbors`, previously a list, is now a dictionary with directions. If wraparound is turned off, cells on the edge will just not have directions that go out of bounds. For example, cells on the north edge will not have `"northwest"`, `"north"`, or `"northeast"` in their `cell.neighbors` dictionary.

Possible changes:
- Add another dictionary to distinguish between a cell's Moore and Von Neumann neighbors. `cell.neighbors` uses whichever neighborhood mode you picked in the config.

### CODE REVIEW

**Comment by colinhanrahan** [comment]
The new logic in `updateNeighbors()` looks just like `cell.findNeighborAgents()`. That function is only called in `doLending()`, which could just use `cell.neighbors` and add a check for `agent != None` like everything else. We can probably remove `findNeighborAgents()` (or if you really want, change the logic of everything else to use this function).

File: agent.py [hunk 1]
```diff
@@ -51,8 +51,7 @@ def __init__(self, agentID, birthday, cell, configuration):
         self.cellsInRange = []
         self.lastMoved = -1
         self.neighborhood = []
-        self.vonNeumannNeighbors = {"north": None, "south": None, "east": None, "west": None}
-        self.mooreNeighbors = {"north": None, "northeast": None, "northwest": None, "south": None, "southeast": None, "southwest": None, "east": None, "west": None}
+        self.socialNeighbors = []
```
Comment (nkremerh) [inline agent.py:54]: Let's just call this `self.neighbors`.

File: agent.py [hunk 2]
```diff
@@ -1319,13 +1300,15 @@ def updateMeanIncome(self, sugarIncome, spiceIncome):
         self.spiceMeanIncome = (alpha * spiceIncome) + ((1 - alpha) * self.spiceMeanIncome)
 
     def updateNeighbors(self):
-        self.updateVonNeumannNeighbors()
-        self.updateMooreNeighbors()
+        self.socialNeighbors = []
+        for neighborCell in self.cell.neighbors.values():
```
Comment (nkremerh) [inline agent.py:1304]: Is there a clean way to do this as a list comprehension one-liner?

File: agent.py (hunk 1 above)
Comment (nkremerh) [reply agent.py:54]: Especially since we have the `updateNeighbors` method for agents.

**Review by nkremerh — CHANGES_REQUESTED** [review]

File: agent.py (hunk 2 above)
Comment (colinhanrahan) [reply agent.py]: Sure!

File: agent.py (hunk 1 above)
Comment (colinhanrahan) [reply agent.py]: Renamed in new commit.

**Review by nkremerh — APPROVED** [review]

**Pull request merged**